libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryEnabler.java

```diff
@@ -19,54 +21,98 @@
     ENABLED, DISABLED
   }
 
-  static final String MAPBOX_SHARED_PREFERENCE_KEY_TELEMETRY_STATE = "mapboxTelemetryState";
+
+  // It is intentional to keep this a Class Level variable, so as to keep its life tied to. Otherwise these listeners are stored as WeakReference's and gets garbage collected over time.
```

Comment:
This might be true for activity, not sure if it's true though for any other class that has more deterministic lifecycle.

Reply:
The idea here is to keep the listener's life tied to the TelemetryEnabler instance irrespective of life cycle of the class where the instance is created. For example, if the TelemetryEnabler instance is created in a Non Activity class, the TelemetryEnabler instance will live as long as the parent class/instance is alive.